### PR TITLE
Fixes #403

### LINKED ISSUE #403: nimble path is inconsistent with nimble develop
For `nimble develop`ed project `nimble path` shows a path to `~/.nimble/pkgs/...`.

### PR DESCRIPTION
**Pull request by yglukhov**

### CODE REVIEW

File: src/nimble.nim [hunk 1]
```diff
@@ -615,15 +615,23 @@ proc listPaths(options: Options) =
...
-
-      if hasSpec:
+      var nimbleFile = path / name.addFileExt("nimble")
+      var nimbleLinkTargetPath: string
+      if not nimbleFile.existsFile:
+        let nimbleLinkFile = path / name.addFileExt("nimble-link")
+        if fileExists(nimbleLinkFile):
+          let lns = readFile(nimbleLinkFile).splitLines()
+          nimbleFile = lns[0]
+          nimbleLinkTargetPath = lns[1]
```
Comment (dom96) [inline src/nimble.nim]: I'm fairly sure there is a function that deals with this logic, please use it instead of duplicating the code.

**Comment by yglukhov** [comment]
> I'm fairly sure there is a function that deals with this logic

Could not find any, so added one.

File: src/nimble.nim [hunk 2]
```diff
@@ -615,15 +615,23 @@ proc listPaths(options: Options) =
       if kind != pcDir or not path.startsWith(options.getPkgsDir / name):
         continue
 
-      let
-        nimbleFile = path / name.addFileExt("nimble")
-        hasSpec = nimbleFile.existsFile
-
-      if hasSpec:
+      var nimbleFile = path / name.addFileExt("nimble")
```
Comment (dom96) [inline src/nimble.nim]: Why don't you just use the `findNimbleFile` proc here?

Comment (yglukhov) [reply src/nimble.nim]: Err. Because we still need to know whether it is a link? Specifically, the `nimbleLinkTargetPath` value.

**Comment by dom96** [comment]
This is what I meant btw: https://github.com/nim-lang/nimble/commit/a02a1431e2167bf2fdf59c25d7a4ed4696c6bbc4

**Comment by yglukhov** [comment]
👍

**Pull request merged**